sycl/source/detail/graph_impl.cpp

```diff
@@ -137,7 +137,8 @@ std::shared_ptr<node_impl> graph_impl::addSubgraphNodes(
         auto Successor = NodesMap[NextNode];
         NodeCopy->registerSuccessor(Successor, NodeCopy);
       } else {
-        assert("Node duplication failed. A duplicated node is missing.");
+        assert(false &&
+               "Node duplication failed. A duplicated node is missing.");
       }
```

Comment:
You don't need to `if` statement in this case. Just move the condition to assert.

```c++
assert(NodesMap.find(NextNode) != NodesMap.end() && "Node duplication failed. A duplicated node is missing.");
auto Successor = NodesMap[NextNode];
NodeCopy->registerSuccessor(Successor, NodeCopy);
```

Reply:
I wasn't aware of this fix because it wasn't linked from the guilty PR, so I've created https://github.com/intel/llvm/pull/10977 that does exactly what @bader asks for.

Reply:
Maybe we can use `at` instead of `operator []` to make out-of-bounds check by std::map and remove assert from the code? NOTE: this will enable the check in Release mode too, which might be not intended.

Reply:
@reble has pushed an update so that this PR uses `at`